cmd/krew/cmd/update.go

```diff
@@ -41,12 +57,147 @@ Remarks:
 	RunE: ensureIndexUpdated,
 }
 
+func retrieveUpdatedPluginList() ([]string, error) {
+	modifiedFiles, err := gitutil.ListModifiedFiles(constants.IndexURI, paths.IndexPluginsPath())
+	if err != nil {
+		return []string{}, err
+	}
+
+	var plugins []string
+	for _, f := range modifiedFiles {
+		filename := filepath.Base(f)
+		extension := filepath.Ext(filename)
+		name := filename[0 : len(filename)-len(extension)]
+		plugins = append(plugins, name)
+	}
+
+	return plugins, nil
+}
+
+func retrievePluginNameVersionMap(names []string) map[string]string {
```

Comment:
I'm not convinced this should be map[string]string.
It's not great for extensibility or readability.

Feel free to either,
- just create a `type pluginInfo struct {name, version string }`, or
- just return the `[]*Plugin` objects.

Reply:
actually it seems like you have a `type newPlugin struct`, so let's follow suit here. either create a new type or find a way of reusing.